### PR TITLE
Practicing pull request: Registering now logs you in as well, slight ui changes for dark mode

### PR DESCRIPTION
**Pull request by Juancarlos140**

### CODE REVIEW

File: resources/js/components/RegisterForm.tsx [hunk 1]
```diff
@@ -33,10 +36,31 @@ export default function RegisterForm({ onRegister }) {
     })
 
     async function register(values: z.infer<typeof formSchema>) {
-        const { data, errors } = await post("/register", values)
+        setIsSubmitting(true);
+        setError(null);
```
Comment (sama305) [inline resources/js/components/RegisterForm.tsx:40]: we never use `error` anywhere else, I think we can remove `error` and `setError` for now

File: resources/js/components/ui/button.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { cva, type VariantProps } from "class-variance-authority"
 import { cn } from "@/lib/utils"
 
 const buttonVariants = cva(
-  "inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-zinc-950 disabled:pointer-events-none disabled:opacity-50 dark:focus-visible:ring-zinc-300",
```
Comment (sama305) [inline resources/js/components/ui/button.tsx:8]: The buttons now are inconsistent with the input fields. Maybe we should keep them rounded..

File: resources/js/components/ui/button.tsx [hunk 3]
```diff
@@ -18,7 +18,7 @@ const buttonVariants = cva(
         secondary:
           "bg-zinc-100 text-zinc-900 shadow-sm hover:bg-zinc-100/80 dark:bg-zinc-800 dark:text-zinc-50 dark:hover:bg-zinc-800/80",
         ghost: "hover:bg-zinc-100 hover:text-zinc-900 dark:hover:bg-zinc-800 dark:hover:text-zinc-50",
-        link: "text-zinc-900 underline-offset-4 hover:underline dark:text-zinc-50",
```
Comment (sama305) [inline resources/js/components/ui/button.tsx:21]: We don't need to change this as shadcn already has a button that looks like this: https://ui.shadcn.com/docs/components/button#ghost. So on the `Navbar` change "Login" and "Register" to have variant `ghost` and only make the corners sharp on those buttons instead of all of them.

File: resources/js/components/ui/form.tsx [hunk 4]
```diff
@@ -93,7 +93,7 @@ const FormLabel = React.forwardRef<
   return (
     <Label
       ref={ref}
-      className={cn(error && "text-red-500 dark:text-red-900", className)}
```
Comment (sama305) [inline resources/js/components/ui/form.tsx:96]: I feel like this might be unnecessary

File: resources/js/components/RegisterDialog.tsx [hunk 5]
```diff
@@ -15,7 +15,7 @@ export function RegisterDialog() {
       <div>
          <Dialog>
           <DialogTrigger asChild>
-            <Button variant='link'>Register</Button>
```
Comment (sama305) [inline resources/js/components/RegisterDialog.tsx:18]: I feel this might be an Unnecessary change

**Review by sama305 — CHANGES_REQUESTED** [review]
The automatic login after registering works, but most of the style changes should be changed. We shouldn't have to override any of the shadcn ui components, so I think those changes might need to be reverted

File: resources/js/components/ui/form.tsx (hunk 4 above)
Comment (Juancarlos140) [reply resources/js/components/ui/form.tsx:96]: I changed this because the text  on the register form (Username, Email, and Password) blended in with the background color, this makes the text white in dark mode when there are no errors

File: resources/js/components/ui/button.tsx (hunk 3 above)
Comment (Juancarlos140) [reply resources/js/components/ui/button.tsx:21]: I reverted the changes I made to the link variant. I made a few changes to the ghost variant to make the style more consistent for dark mode

File: resources/js/components/ui/button.tsx (hunk 2 above)
Comment (Juancarlos140) [reply resources/js/components/ui/button.tsx]: I reverted the changes

File: resources/js/components/RegisterForm.tsx (hunk 1 above)
Comment (Juancarlos140) [reply resources/js/components/RegisterForm.tsx]: I removed setError and removed all the throw and catch conditions

**Review by alex-305 — APPROVED** [review]
